src/Bridges/Constraint/soc_to_nonconvex_quad.jl

```diff
@@ -0,0 +1,195 @@
+abstract type AbstractSOCtoNonConvexQuadBridge{T} <: AbstractBridge end
+
+"""
+    SOCtoNonConvexQuadBridge{T}
+
+Constraints of the form `VectorOfVariables`-in-`SecondOrderCone` can be
+transformed into a `ScalarQuadraticFunction`-in-`LessThan` and a
+`ScalarAffineFunction`-in-`GreaterThan`. Indeed, the definition of the
+second-order cone
+```math
+t \\ge \\lVert x \\rVert_2 \\  (1)
+```
+is equivalent to
+```math
+\\sum x_i^2 \\le t^2  (2)
+```
+with ``t \\ge 0``.  (3)
+
+*WARNING* This transformation starts from a convex constraint (1) and creates a
+non-convex constraint (2), because the Q matrix associated with the constraint 2
+has one negative eigenvalue. This might be wrongly interpreted by a solver.
+Some solvers can look at (2) and understand that it is a second order cone, but
+this is not a general rule.
+For these reasons this bridge is not automatically added by full_bridge_optimizer.
+Care is recommended when adding this bridge to a optimizer.
+"""
+struct SOCtoNonConvexQuadBridge{T} <: AbstractSOCtoNonConvexQuadBridge{T}
+    quad::CI{MOI.ScalarQuadraticFunction{T}, MOI.LessThan{T}}
+    var_pos::Vector{CI{MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}}}
+    vars::Vector{MOI.VariableIndex}
+end
+function bridge_constraint(::Type{SOCtoNonConvexQuadBridge{T}}, model,
+                           func::MOI.VectorOfVariables,
+                           set::MOI.SecondOrderCone) where T
+
+    vis = func.variables
+
+    t = vis[1]
+    x = vis[2:end]
+    a_terms = MOI.ScalarAffineTerm{T}[]
+    q_terms = MOI.ScalarQuadraticTerm{T}[]
+    push!(q_terms, MOI.ScalarQuadraticTerm(-T(2), t, t))
+    for var in x
+        push!(q_terms, MOI.ScalarQuadraticTerm(T(2), var, var))
+    end
+
+    fq = MOI.ScalarQuadraticFunction(a_terms, q_terms, zero(T))
+    quad = MOI.add_constraint(model, fq, MOI.LessThan(zero(T)))
+    # ScalarAffineFunction's are added instead of SingleVariable's
+    # because models can only have one SingleVariable per variable
```

Comment:
This is a bit unclear, as the bridge does not any other bounds. Clarify that this is in case other bounds were already added in the model outside of the bridge.